test/src/test/integration/link_git_protocol.rs

```diff
@@ -341,6 +363,8 @@ where
         update_tips(&local_repo, &out.wanted_refs).unwrap();
     }
 
+    // Need to refresh it as it didn't notice the new pack
```

Comment:
Interesting.. what does it exactly mean to `open` a repository, in terms of I/O, file handles, etc? 

I'm asking because `git2` does a **lot** actually (including resolving and parsing config files), so we ended up using `git2::Repository` from a (database) pool to avoid fd exhaustion and other issues. I suspect that we'd not use easy mode in a lot of places in order to be able to carefully manage resource consumption, but I think documentation about what the cost of an easy repo is would help.

Reply:
Indeed, documentation is non-existing in places and I plan to rectify that with a quick-pass of doc-writing before I move on.

Right now `open(…)` doesn't do much except for mapping all packs in full and reading the local repository config file once. Everything else is lazy.

On the server you would do the same, but instead of holding a `git::Repository` you would hold a `git::EasyArcExclusive`. That way, access to the persistent pack memory maps is shared, but caches and on-demand maps (like packed-refs) are not.

Even though it's not implemented, I would like to have a `refresh()` to reload the object database. And thinking about it, it's easy to demo this API here but would then force the `local` repo to be an `EasyArcExclusive` as well.

Probably worth sketching this out while we are here.

Note that `git-config` will be fully loaded into memory so no handles will be left open.